bin/pulsar

```diff
@@ -290,6 +290,9 @@ if [[ -z "$IS_JAVA_8" ]]; then
   OPTS="$OPTS --add-opens java.base/sun.net=ALL-UNNAMED --add-opens java.management/sun.management=ALL-UNNAMED"
 fi
 
+# Specify default nar tmp dir
+OPTS="$OPTS -Dnar.extraction.tmpdir=$PULSAR_HOME/tmp"
```

Comment:
Please don't change this default behaviour, we are going to break existing users that upgrade.

Now that you hBe the system property you can let your users configure it

Reply:
Default use the /tmp dir has a risk which will cause the class file be auto cleaned by os, when the class file 
 used again , the class not find error occur . Now when the `nar.extraction.tmpdir` not configured will use the old config `java.io.tmpdir` , so the change will not break existing users again .

Reply:
there is no guarantee that $PULSAR_HOME/tmp exists and it is writable by the user.
if we change the behaviour then we have to document it, but, as a general rule, we cannot break things during upgrades.
it is better to add this feature, in 2.11, then we can change the default behaviour in 2.12.

also, in linux /tmp is not emptied randomly, my experience it that such directory is empty at boot

Reply:
> it is better to add this feature, in 2.11, then we can change the default behaviour in 2.12.

Thx for the Suggestions.

> also, in linux /tmp is not emptied randomly, my experience it that such directory is empty at boot

maybe diff linux os has diff behaviours for the /tmp dir clean, Debian will clean when the boot , but as Centos or RedHat  will  auto clean the file in the /tmp dir when  7 days or 30 days the file not be used, In this way, the .class file will loss. 
In our case, we add the KOP protocol, after few days when we want use the KOP protocol, we find  some .class have been cleaned by os.